ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e: Initial AWS UPI template

wking opened the pull request:
Similar to #3305, but this template is AWS-specific.  I'll look into unifying later.  I also still need to wire up an installer job for this.

CC @abhinavdahiya, @cuppett, @staebler, @vrutkovs

vrutkovs commented on ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml:
```diff
@@ -0,0 +1,611 @@
+          # TODO: 06_cluster_worker_node.yaml
+        fi
+
+        # TODO: e2e suite
```
e2e suite is running in `test` container, it would start once setup is complete

wking replied:
> e2e suite is running in `test` container...

:man_facepalming: Dropped in dac4d74a4 -> 007745d53.

vrutkovs commented on ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml:
```diff
@@ -0,0 +1,611 @@
+
+          echo "Deprovisioning cluster ..."
+          export AWS_SHARED_CREDENTIALS_FILE=/etc/openshift-installer/.awscred
+          openshift-install --dir /tmp/artifacts/installer destroy cluster
```
Does `destroy cluster` also remove UPI infra?

wking replied:
> Does destroy cluster also remove UPI infra?

Yes for AWS, because I'm setting `--tags` on the stacks, which tags both the stack and resources created for that stack.  I've just opened openshift/installer#1595 to handle removing stacks that match the tags listed in `metadata.json` as as the cluster resources, so we should be good to go here.

wking commented:
[e2e-aws-upi][1]:

```
2019/04/11 10:42:37 Running pod e2e-aws-upi
Installing from release registry.svc.ci.openshift.org/ci-op-xqpc1pw7/release@sha256:56f0603347fb1e71596953ae1b4a637ab89dce8be8e268ca8cef8646631a3f22
level=info msg="Consuming \"Install Config\" from target directory"
level=warning msg="Found override for ReleaseImage. Please be warned, this is not advised"
/bin/sh: line 80: aws: command not found
```

No `aws`?  We [ask for it][2].  Checking the image myself:

```console
$ podman pull registry.svc.ci.openshift.org/ci-op-xqpc1pw7/release@sha256:56f0603347fb1e71596953ae1b4a637ab89dce8be8e268ca8cef8646631a3f22
$ podman run --rm -it --entrypoint rpm registry.svc.ci.openshift.org/ci-op-xqpc1pw7/release@sha256:56f0603347fb1e71596953ae1b4a637ab89dce8be8e268ca8cef8646631a3f22 -q awscli
package awscli is not installed
```

Can we get the build log from somewhere?  Maybe this aspect of things doesn't play nicely with rehearsals?  Wait...

```console
$ podman inspect --format '{{.Config.Entrypoint}}' registry.svc.ci.openshift.org/ci-op-xqpc1pw7/release@sha256:56f0603347fb1e71596953ae1b4a637ab89dce8be8e268ca8cef8646631a3f22
[/usr/bin/cluster-version-operator]
```

how did a CVO image get over here?

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/2/build-log.txt
[2]: https://github.com/openshift/installer/blob/1ed3e285ec915a0d5f76a2a25fd9773babd850af/images/installer/Dockerfile.upi.ci#L28

vrutkovs commented:
> No `aws`? We [ask for it](https://github.com/openshift/installer/blob/1ed3e285ec915a0d5f76a2a25fd9773babd850af/images/installer/Dockerfile.upi.ci#L28). Checking the image myself:

https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/2/artifacts/build-logs/upi-installer.log.gz says `No package epel-release available.` and thus no awscli.
You'd need to contact ART to include it in base repos (and drop epel)

wking commented:
Rebased and updated with 799fe4a7b -> 71cdc8977, although in hindsight I implemented under the assumption that openshift/installer#1649 had landed :p.

vrutkovs commented on ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml:
```diff
@@ -363,69 +385,248 @@ objects:
+            compute_ignition = <<END_OF_WORKER_IGNITION
+            ${WORKER_IGN}
+            END_OF_WORKER_IGNITION
+          EOF
```
```suggestion
        EOF
```
This seems to break CI test

wking replied:
Fixed the here-doc indentation (I think ;) with 422d57100 -> ef62451ab.

wking commented:
Hooray, [vSphere is now up to][1]:

```
Apply complete! Resources: 31 added, 0 changed, 0 destroyed.
Waiting for bootstrap to complete
level=info msg="Waiting up to 30m0s for the Kubernetes API at https://api.ci-op-gr07n873-dbc3b.origin-ci-int-aws.dev.rhcloud.com:6443..."
level=info msg="Use the following commands to gather logs from the cluster"
level=info msg="openshift-install gather bootstrap --help"
level=fatal msg="waiting for Kubernetes API: context deadline exceeded"
```

I'm trying to figure out if that's vSphere's current standard...

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-vsphere/4

abhinavdahiya commented on ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml:
```diff
@@ -363,18 +385,188 @@ objects:
-        mkdir -p /tmp/tf
+        # begin bootstrapping
+        if [[ "${CLUSTER_TYPE}" == "aws" ]]; then
+          RHCOS_AMI=ami-0df3f99538fbef10f  # FIXME: assumes AWS_REGION is us-east-1
```
you can use something similar to https://github.com/openshift/release/pull/3612

wking replied:
> you can use something similar to #3612

Are we committed to that approach?  Seems [like a hack][1] to me.  And if so, this isn't much more of a hack ;).

[1]: https://github.com/openshift/installer/pull/1673#issuecomment-486692546

abhinavdahiya replied:
> Are we committed to that approach? Seems like a hack to me. And if so, this isn't much more of a hack ;).
That's only for our testing of UPI. and using the copied rhcos.json atleast makes sure we don't need to bump bootsimages here when we bump it in installer.

wking replied:
> That's only for our testing of UPI...

Right, but holding out for a real fix will help motivate us to give customers something so that _they_ don't have to bump bootimages when we bump them in the installer ;).  Dog food, for the win :p

abhinavdahiya replied:
> > That's only for our testing of UPI...
> 
> Right, but holding out for a real fix will help motivate us to give customers something so that _they_ don't have to bump bootimages when we bump them in the installer ;). Dog food, for the win :p

but we cannot test the bump in installer if you hard code here.

wking replied:
> but we cannot test the bump in installer if you hard code here.

ahh, true.  But we don't have tight doc coupling now, so there's some benefit in testing both the old AMI (which the docs will still be recommending) and the new AMI (which IPI will use).

droslean commented:
/test pj-rehearse

wking commented:
[e2e-aws-upi][1]:

```
An error occurred (AccessDenied) when calling the CreateStack operation: User: arn:aws:iam::460538899914:user/origin-ci-robot-provision is not authorized to perform: cloudformation:CreateStack on resource: arn:aws:cloudformation:us-east-1:460538899914:stack/ci-op-mk19233w-38af6-vpc/*
```

So I've attached full access to `cloudformation:*` to that user.

/retest

wking commented:
Rollbacks because:

```
The load balancer name 'api-int-ci-op-mk19233w-38af6-us-east-1' cannot be longer than '32' characters (Service: AmazonElasticLoadBalancingV2; Status Code: 400; Error Code: ValidationError; Request ID: 2acccf34-6d74-11e9-ac17-cf3d0e8e1c6d)
```

I've filed openshift/installer#1706 to address that.

wking commented:
```
User: arn:aws:iam::460538899914:user/origin-ci-robot-provision is not authorized to perform: lambda:CreateFunction on resource: arn:aws:lambda:us-east-1:460538899914:function:ci-38af6-infra-RegisterSubnetTags-BMYO0XZJYDEU (Service: AWSLambdaInternal; Status Code: 403; Error Code: AccessDeniedException; Request ID: 41c9626a-6d7c-11e9-861e-81cbcb0a03e0)
```

So I've added the AWSLambdaFullAccess policy to the CI user.

wking commented:
/test e2e-aws-upi

wking commented:
/retest

wking commented:
It will be a bit until [the e2e-aws-upi job][1] gets torn down and reported here, but the results are:

```

Flaky tests:

[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support existing directories when readOnly specified in the volumeSource [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support non-existent path [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] volumes should allow exec of files on the volume [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] Volume expand Verify if editing PVC allows resize [Suite:openshift/conformance/parallel] [Suite:k8s]

Failing tests:

[sig-network] Networking should provide Internet connection for containers [Feature:Networking-IPv4] [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] Multi-AZ Clusters should spread the pods of a replication controller across zones [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] Multi-AZ Clusters should spread the pods of a service across zones [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] provisioning should provision storage with mount options [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support existing single file [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support file as subpath [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support readOnly directory specified in the volumeMount [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support readOnly file specified in the volumeMount [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (ext3)] volumes should allow exec of files on the volume [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (filesystem volmode)] volumeMode should create sc, pod, pv, and pvc, read/write to the pv, and delete all created resources [Suite:openshift/conformance/parallel] [Suite:k8s]
```

with a successful install :tada:.  Still a few FIXMEs to sort out.  I'm fine either landing this as it stands and iterating on those, or holding this open until I've ground them off.

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/16

wking commented:
Let's see if those failures are general flakes or something missing from the UPI approach:

/retest

staebler commented:
/test pj-rehearse

staebler commented:
/test pj-rehearse

staebler commented:
This PR looks good to me, as far as not breaking e2e-vsphere tests.

wking commented:
[e2e-aws][1]:

```
level=fatal msg="failed to initialize the cluster: Some cluster operators are still updating: authentication, console, image-registry: timed out waiting for the condition"
```

[e2e-vsphere][2]:

```
Failing tests:

[Feature:Platform] Managed cluster should have no crashlooping pods in core namespaces over two minutes [Suite:openshift/conformance/parallel]
```

/retest

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/21
[2]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-vsphere/19

staebler commented:
The e2e-vsphere tests fail that that particular test fairly consistently across PRs. It is not a failure brought on by the changes in this PR.

wking commented:
With 136ab5408 -> 7b2c175fb, I've rebased onto master, dropped the openshift/installer#1706 workarounds now that that's landed, and pushed a compute node into the second private subnet to [avoid failing][1]:

```
[sig-scheduling] Multi-AZ Clusters should spread the pods of a service across zones [Suite:openshift/conformance/parallel] [Suite:k8s]
```

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/22

wking commented:
[e2e-aws][1]:

```
level=fatal msg="failed to initialize the cluster: Cluster operator image-registry is still updating: timed out waiting for the condition"
```

Looking at the ClusterOperator:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/23/artifacts/e2e-aws-upi/clusteroperators.json | jq -r '.items[] | select(.metadata.name == "image-registry") | ([.status.conditions[] | {key: .type, value: .}] | from_entries).Progressing'
{
  "lastTransitionTime": "2019-05-04T03:45:29Z",
  "message": "All resources are successfully applied, but the deployment does not exist",
  "reason": "WaitingForDeployment",
  "status": "True",
  "type": "Progressing"
}
```

Checking that Deployment:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/23/artifacts/e2e-aws-upi/must-gather/namespaces/openshift-image-registry/apps/deployments/image-registry.yaml | yaml2json | jq .status.conditions
[
  {
    "reason": "MinimumReplicasAvailable",
    "message": "Deployment has minimum availability.",
    "type": "Available",
    "status": "True",
    "lastTransitionTime": "2019-05-04T03:45:48Z",
    "lastUpdateTime": "2019-05-04T03:45:48Z"
  },
  {
    "reason": "NewReplicaSetAvailable",
    "message": "ReplicaSet \"image-registry-7749f787d4\" has successfully progressed.",
    "type": "Progressing",
    "status": "True",
    "lastTransitionTime": "2019-05-04T03:45:29Z",
    "lastUpdateTime": "2019-05-04T03:45:48Z"
  }
]
```

I dunno why we took so long we timed out (by 19 seconds!?).  But whatever, I'll just kick it again out of curiousity:

/retest

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/23

wking commented:
Hrm, same [e2e-aws error][1] as [last time][2].  This time:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/24/artifacts/e2e-aws-upi/installer/.openshift_install.log | grep fatal
time="2019-05-04T05:42:27Z" level=fatal msg="failed to initialize the cluster: Cluster operator image-registry is still updating: timed out waiting for the condition"
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/24/artifacts/e2e-aws-upi/must-gather/cluster-scoped-resources/config.openshift.io/clusterversions.yaml | yaml2json | jq '.items[0].status.conditions[] | select(.type == "Failing" or .type == "Progressing")'
{
  "lastTransitionTime": "2019-05-04T05:37:08Z",
  "message": "Cluster operator image-registry is still updating",
  "status": "True",
  "type": "Failing",
  "reason": "ClusterOperatorNotAvailable"
}
{
  "lastTransitionTime": "2019-05-04T05:07:58Z",
  "message": "Unable to apply 0.0.1-2019-05-04-044538: the cluster operator image-registry has not yet successfully rolled out",
  "status": "True",
  "type": "Progressing",
  "reason": "ClusterOperatorNotAvailable"
}
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/24/artifacts/ee-aws-upi/must-gather/cluster-scoped-resources/config.openshift.io/clusteroperators/image-registry.yaml | yaml2json | jq .status.conditions
[
  {
    "status": "False",
    "message": "The deployment does not have available replicas",
    "lastTransitionTime": "2019-05-04T05:13:32Z",
    "reason": "NoReplicasAvailable",
    "type": "Available"
  },
  {
    "status": "True",
    "message": "The deployment has not completed",
    "lastTransitionTime": "2019-05-04T05:13:32Z",
    "reason": "DeploymentNotCompleted",
    "type": "Progressing"
  },
  {
    "status": "False",
    "lastTransitionTime": "2019-05-04T05:13:32Z",
    "type": "Degraded"
  }
]
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/24/artifacts/e2e-aws-upi/pods/openshift-image-registry_cluster-image-registry-operator-fd4799767-9gfvh_cluster-image-registry-operator.log | gunzip | grep Unable
I0504 05:13:39.999971       1 controller.go:199] object changed: *v1.Config, Name=cluster (status=true): changed:status.conditions.5.message={"The deployment has not completed" -> "Unable to apply resources: unable to sync storage configuration: exactly one storage type should be configured at the same time, got 2: [EmptyDir S3]"}, changed:status.conditions.5.reason={"DeploymentNotCompleted" -> "Error"}, changed:status.observedGeneration={"2.000000" -> "3.000000"}
```

But I see no [`configs.imageregistry.operator.openshift.io`][3] under [here][4].  Buggy defaulting?

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/24
[2]: https://github.com/openshift/release/pull/3440#issuecomment-489293528
[3]: https://github.com/openshift/cluster-image-registry-operator/blob/fe26b2b64b36becfea87406c9f37ad2ae947a0de/manifests/00-crd.yaml#L5
[4]: https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/24/artifacts/e2e-aws-upi/must-gather/cluster-scoped-resources/operator.openshift.io/

staebler commented:
@wking Is it correct for the image-registry storage to be set to empty dir for UPI aws?

If I am reading the code correctly, the image-registry operator sets the storage to S3 [here](https://github.com/openshift/cluster-image-registry-operator/blob/b27a04e9ffbab9e6ab75de911dbf8acd2befdea4/pkg/storage/storage.go#L96-L97). While the CI tests set the image-registry storage to empty directory [here](https://github.com/openshift/release/blob/7b2c175fb7432be74bea424133df901f3de67eea/ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml#L663).

wking commented:
> Is it correct for the image-registry storage to be set to empty dir for UPI aws?

Nope, good catch.  Fixed with 7b2c175fb -> 7d4e4349b, which will leave the registry provisioning its own S3 bucket.  Folks taking the UPI path may want that, or they may want to configure the registry to use an existing S3 bucket that they create themselves.  For now, a registry-provisioned bucket seems easiest to exercise in CI.

wking commented:
Ok, this round, [AWS[1]:

```
Flaky tests:

[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support existing directory [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] Volume expand Verify if editing PVC allows resize [Suite:openshift/conformance/parallel] [Suite:k8s]

Failing tests:

[sig-network] Networking should provide Internet connection for containers [Feature:Networking-IPv4] [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] Multi-AZ Clusters should spread the pods of a replication controller across zones [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] Multi-AZ Clusters should spread the pods of a service across zones [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support non-existent path [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (ext3)] volumes should allow exec of files on the volume [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (ext3)] volumes should be mountable [Suite:openshift/conformance/parallel] [Suite:k8s]
```

[vSphere][2]:

```
level=fatal msg="failed to initialize the cluster: Cluster operator machine-config is reporting a failure: Failed to resync 0.0.1-2019-05-06-182506 because: timed out waiting for the condition during waitForDaemonsetRollout: Daemonset machine-config-daemon is not ready. status: (desired: 6, updated: 6, ready: 5, unavailable: 5): timed out waiting for the condition"
```

Still to `Multi-AZ Clusters should spread the pods of a ... across zones` issue; not sure what's going on there now that I'm using compute in two zones just like the IPI tests.

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/25
[2]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-vsphere/23

wking commented:
Double checking the node locations, we get the expected distribution:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/25/artifacts/e2e-aws-upi/nodes.json | jq '.items[] | {name: .metadata.name, zone: .metadata.labels["failure-domain.beta.kubernetes.io/zone"], conditions: ([.status.conditions[] | {key: .type, value: .status}] | from_entries)}'
{
  "name": "ip-10-0-50-219.ec2.internal",
  "zone": "us-east-1a",
  "conditions": {
    "MemoryPressure": "False",
    "DiskPressure": "False",
    "PIDPressure": "False",
    "Ready": "True"
  }
}
{
  "name": "ip-10-0-62-49.ec2.internal",
  "zone": "us-east-1a",
  "conditions": {
    "MemoryPressure": "False",
    "DiskPressure": "False",
    "PIDPressure": "False",
    "Ready": "True"
  }
}
{
  "name": "ip-10-0-69-73.ec2.internal",
  "zone": "us-east-1b",
  "conditions": {
    "MemoryPressure": "False",
    "DiskPressure": "False",
    "PIDPressure": "False",
    "Ready": "True"
  }
}
{
  "name": "ip-10-0-75-186.ec2.internal",
  "zone": "us-east-1b",
  "conditions": {
    "MemoryPressure": "False",
    "DiskPressure": "False",
    "PIDPressure": "False",
    "Ready": "True"
  }
}
{
  "name": "ip-10-0-91-97.ec2.internal",
  "zone": "us-east-1c",
  "conditions": {
    "MemoryPressure": "False",
    "DiskPressure": "False",
    "PIDPressure": "False",
    "Ready": "True"
  }
}
```

wking commented:
:man_shrugging:

/retest

wking commented:
[AWS][1]:

```
level=fatal msg="failed to initialize the cluster: Some cluster operators are still updating: authentication, console: timed out waiting for the condition"
```things

/retest

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/26

wking commented:
[AWS][1]:

```
Flaky tests:

[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] provisioning should provision storage with defaults [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support existing single file [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (filesystem volmode)] volumeMode should create sc, pod, pv, and pvc, read/write to the pv, and delete all created resources [Suite:openshift/conformance/parallel] [Suite:k8s]

Failing tests:

[sig-network] Networking should provide Internet connection for containers [Feature:Networking-IPv4] [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] Multi-AZ Clusters should spread the pods of a replication controller across zones [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] Multi-AZ Clusters should spread the pods of a service across zones [Suite:openshift/conformance/parallel] [Suite:k8s]
```

Those happened [last time][2] too, so they're probably real.  Still digging...

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/27
[2]: https://github.com/openshift/release/pull/3440#issuecomment-489778334

sdodson commented:
/retest

wking commented:
[e2e-aws][1] has the same multi-AZ errors as before:

```

Failing tests:

[sig-network] Networking should provide Internet connection for containers [Feature:Networking-IPv4] [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] Multi-AZ Clusters should spread the pods of a replication controller across zones [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] Multi-AZ Clusters should spread the pods of a service across zones [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: aws] [Testpattern: Dynamic PV (default fs)] subPath should support existing directories when readOnly specified in the volumeSource [Suite:openshift/conformance/parallel] [Suite:k8s]
```

Still dunno what's going on there.

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3440/rehearse-3440-pull-ci-openshift-installer-master-e2e-aws-upi/28

sdodson commented:
/lgtm

wking commented:
Pushed 7d4e4349b -> 9ab4e372e, fixing [the generated-config error][1] and adding an additional commit to fix that issue for the other presubmit jobs too (following earlier partial work in #3713).  @sdodson, re-`/lgtm`?

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3440/pull-ci-openshift-release-master-generated-config/5088/build-log.txt

sdodson commented:
/lgtm

wking commented:
I think #3775 will fix the multi-zone errors.
